numba/core/interpreter.py

```diff
@@ -79,6 +79,488 @@ def get_assignment_source(self, destname):
+        $varargs_var = $final_tuple
+
+    It unclear if the extra assignment at the end is always present.
+    In addition to collecting and returning the original args, we also
+    delete any IR statments that uses any of the tuples from new_body.
+    """
+    errmsg = "CALL_FUNCTION_EX with **kwargs not supported"
+    # We traverse to the front of the block to look for the original
+    # tuple.
+    search_start = 0
+    total_args = []
+    if (
+        isinstance(vararg_stmt, ir.Assign)
+        and isinstance(vararg_stmt.value, ir.Var)
+    ):
+        target_name = vararg_stmt.value.name
+        # If there is an initial assignment, delete it
+        new_body[search_end] = None
+        search_end -= 1
+    else:
+        target_name = vararg_stmt.target.name
+    start_not_found = True
+    # Traverse backwards to find all concatentations
+    # until eventually reaching the original empty tuple.
+    while search_end >= search_start and start_not_found:
+        concat_stmt = body[search_end]
+        if (
+            isinstance(concat_stmt, ir.Assign)
+            and concat_stmt.target.name == target_name
+            and isinstance(concat_stmt.value, ir.Expr)
+            and concat_stmt.value.op == "build_tuple"
+            and not concat_stmt.value.items
+        ):
+            # If we have reached the build_tuple
+            # we exit.
+            start_not_found = False
+            new_body[search_end] = None
+        else:
+            # We expect to find another arg to append.
+            # The first stmt must be an add
+            if (search_end == search_start) or not (
+                isinstance(concat_stmt, ir.Assign)
+                and (
+                    concat_stmt.target.name
+                    == target_name
+                )
+                and isinstance(
+                    concat_stmt.value, ir.Expr
+                )
+                and concat_stmt.value.op == "binop"
+                and concat_stmt.value.fn == operator.add
+            ):
+                # We cannot handle this format.
+                raise UnsupportedError(errmsg)
+            lhs_name = concat_stmt.value.lhs.name
+            rhs_name = concat_stmt.value.rhs.name
+            # The previous statment should be a
+            # build_tuple containing the arg.
+            arg_tuple_stmt = body[search_end - 1]
+            if not (
+                isinstance(arg_tuple_stmt, ir.Assign)
+                and isinstance(
+                    arg_tuple_stmt.value, ir.Expr
+                )
+                and (
+                    arg_tuple_stmt.value.op
+                    == "build_tuple"
+                )
+                and len(arg_tuple_stmt.value.items) == 1
+            ):
+                # We cannot handle this format.
+                raise UnsupportedError(errmsg)
+            if arg_tuple_stmt.target.name == lhs_name:
+                target_name = rhs_name
```

Comment:
Impossible case? cpython doesn't seem to put the tuple on the LHS.

Reply:
I agree I also never saw this on the LHS. I was simply trying to be conservative against any slight changes in the generated code. Since this is a property of the bytecode should I just remove this case?

Reply:
Let's make this raise an error indicating it is unreachable/impossible.